tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -2925,63 +2922,62 @@ private[optimizer] abstract class OptimizerCore(
       }
     }
 
-    val info = new LabelInfo(newLabel, acceptRecords = usePreTransform)
-    withState(info.returnedTypes) {
-      val bodyScope = scope.withEnv(scope.env.withLabelInfo(oldLabelName, info))
+    val info = new LabelInfo(newLabel, acceptRecords = usePreTransform,
+        returnedTypes = newSimpleState(Nil))
```

Comment:
Using a named arg for `returnedTypes` here is inconsistent with what you use above for `LabelInfo`. Is this intentional?